test/e2e/functional/tests/e2e.go

```diff
@@ -3,355 +3,918 @@ package tests
 )
 
-func init() {
-	if len(os.Getenv("IS_OPENSHIFT")) != 0 {
-		TestIsOpenShift = true
+// testIngressNodeFirewall represents one IngressNodeFirewall object and is used to generate the sourceCIDRs and protocol
+// rules.
+type testIngressNodeFirewall struct {
+	testRules []testRule
+}
+
+// testRule is used as a template to generate sourceCIDRs entries and protocol rules
+type testRule struct {
+	// sourceCIDRsEntry is used to generate an INFs ingress rule sourceCIDRs. The number of entries equals the number of
+	// sourceCIDRs created. It must contain at least one pod name from which IPV4 and IPV6 CIDRs are generated.
+	sourceCIDRsEntries []sourceCIDRsEntry
+	// protoRules is a list of funcs that return a proto rule for an INFs ingress rule.
+	// These funcs serves as templates to generate rules for one or more protocols.
+	// The list of protocols passed to the function depends on the test description.
+	// Order is generated by the test harness to be unique for a source CIDR.
+	protoRules []func(proto ingressnodefwv1alpha1.IngressNodeFirewallRuleProtocolType, order uint32) ingressnodefwv1alpha1.IngressNodeFirewallProtocolRule
+}
+
+// sourceCIDRsEntry is used to generate a sourceCIDR for an INF ingress rule
+// For example, if a given podName pods IPV4 address ends up being 172.168.1.8 with a v4CIDRPrefix of 24, then the generated
+// sourceCIDR for IPV4 will be 172.168.1.0/24. In order to ensure IPV6 test coverage, its mandatory to also define
+// v6CIDRPrefix.
+type sourceCIDRsEntry struct {
+	podName      string
+	v4CIDRPrefix string
+	v6CIDRPrefix string
+}
+
+// reachable is used by the test code to determine what connectivity we expect or not expect.
+type reachable struct {
+	// from defines the pod name which is the client in all test scenarios. Packets are generated from this client.
+	from string
+	// to defines the pod name which is the server in all test scenarios. Packets are terminated for transport protocol
+	// test cases. For ICMP protocol test cases, "to" pod is used to set up networking and the server does nothing.
+	to string
+	// port is used only for transport protocol tests and is ignored for ICMP test cases
+	port string
+	// connectivity informs the test code what connectivity we expect during the test case
+	connectivity bool
+}
+
+type ips struct {
+	v4 string
+	v6 string
+}
+
+func (i *ips) setV4IP(ips []corev1.PodIP) {
+	for _, ip := range ips {
+		parsedIP := net.ParseIP(ip.IP)
+		if parsedIP.To4() != nil {
+			i.v4 = parsedIP.String()
+		}
 	}
+}
+
+func (i *ips) setV6IP(ips []corev1.PodIP) {
+	for _, ip := range ips {
+		parsedIP := net.ParseIP(ip.IP)
+		if parsedIP.To4() == nil {
+			i.v6 = parsedIP.String()
+		}
+	}
+}
+
+func init() {
 	if ns := os.Getenv("OO_INSTALL_NAMESPACE"); len(ns) != 0 {
 		OperatorNameSpace = ns
 	}
 	if intf := os.Getenv("NODE_INTERFACE"); len(intf) != 0 {
-		TestInterface = intf
+		testInterface = intf
 	}
-	if len(os.Getenv("IS_SINGLESTACK")) != 0 {
-		TestIsSingleStack = true
```

Comment:
hmm those env var are in use by the CI why did u remove them ?

Reply:
CI is always dual stack so i didnt see the need for it.

Reply:
We used to debug ci on an existing cluster that was provisioned as single stack

Reply:
Added back `IS_SINGLESTACK` option.